build_tools/cmake/iree_check_test.cmake

```diff
@@ -140,6 +142,72 @@ function(iree_check_test)
   endif()
 
   iree_package_name(_PACKAGE_NAME)
+
+  # If the source file is from the web, create a custom command to download
+  # it and wrap that with a custom target so later we can use for dependency.
+  if("${_RULE_SRC}" MATCHES "^https?://")
+    set(_SRC_URL "${_RULE_SRC}")
+    # Update the source file to the downloaded-to place.
+    string(REPLACE "/" ";" _SRC_URL_SEGMENTS "${_SRC_URL}")
+    list(POP_BACK _SRC_URL_SEGMENTS _LAST_URL_SEGMENT)
+    set(_DOWNLOAD_TARGET "${_PACKAGE_NAME}_${_LAST_URL_SEGMENT}")
```

Comment:
Core tests must not have dependencies on the network. This is running at build time and not test time, but we still need to be very careful about how this is implemented, and I hesitate to allow this at all in the core project (see also my other comment). In particular, we'd want to make sure this does downloads the latest version of a file only if the local version is out of date. We might even want to download at configure time if a CMake option is set, or download using a script as a manual step prior to using CMake.

We might also want
* an allowlist of URLs for the download script to consider, to ensure that only trusted sources are used
* a list of mirror URLs to try in case the original is not available